src/lib/locale/locales.js

```diff
@@ -50,21 +50,12 @@ function loadLocale(name) {
     // TODO: Find a better way to register and load all the locales in Node
     if (!locales[name] && (typeof module !== 'undefined') &&
             module && module.exports) {
-        oldLocale = globalLocale._abbr;
         try {
-            require('moment/locale/' + name);
-        } catch (e) {
-            // In the test environment, the external module 'moment'
-            // can't be resolved because we're running inside it.
-            // Fallback to using the old relative import
-            try {
-                require('./locale/' + name);
-            } catch (e) { }
-        }
-
-        // because defineLocale currently also sets the global locale, we
-        // want to undo that for lazy loaded locales
-        getSetGlobalLocale(oldLocale);
+            oldLocale = globalLocale._abbr;
+            var aliasedRequire = require;
```

Comment:
Why is it aliased?

Reply:
TBH I don't know *why* this makes metro-bundler work for React native, I just know that it *does* do that.

Reply:
This is still not working for a lot people because React Native is referencing moment-with-locales.min.js i.e. the minified version.  For some reason, the minified version does not seem to have the fix.  

When I fixed the code in moment-with-locales.min.js using your approach of aliasing "require", 2.19.1 began to work for me.